### ISSUE TITLE
Console App failed when receiving status 500

### ISSUE THREAD

**Issue by kerdrek** [LanceMcCarthy/RemixDownloader#17]
I launch the console application on the default xbox user id and when the server respond 500 (Internal Server Error) the app stop, and I have to restart the download from the beginning

Looks like the line 94 of Program.cs in RemixDownloader.Console
`result = await RemixApiService.Current.GetModelsForUserAsync(userId, contUrl);`
is throwing an exception if the HTTP status of the request is not 200

Stack trace

320 models completed...
Unhandled Exception: System.Net.Http.HttpRequestException: Response status code does not indicate success: 500 (Internal Server Error).
at System.Net.Http.HttpResponseMessage.EnsureSuccessStatusCode()
at System.Net.Http.HttpClient.GetStringAsyncCore(Task`1 getTask)
at RemixDownloader.Core.Services.RemixApiService.GetModelsForUserAsync(String userId, String continuationUrl) in C:\Users\...\source\repos\RemixDownloader\src\RemixDownloader\RemixDownloader.Core\Services\RemixApiService.cs:line 58
at RemixDownloader.Console.Program.Main(String[] args) in C:\Users\...\source\repos\RemixDownloader\src\RemixDownloader\RemixDownloader.Console\Program.cs:line 94
at RemixDownloader.Console.Program.<Main>(String[] args)

**Comment by LanceMcCarthy** [comment]
Thank you for reporting this, I'm going to add an additional nested `continue` so that this doesn't stop your downloads. A 500 could be caused my many reasons, I'll also add some additional logic to output the API's header when it returns a non-success.

It would be frustrating to have to start all over again when it breaks halfway through. Unfortunately, there's nothing in the API that would let me say "Start at the 320th model".

**Comment by LanceMcCarthy** [comment]
I pushed a fix to master. In case there is a transient issue getting any set of models, it will write the error out, but still try again.

It's entirely possible that this may end up in a permanent loop and you'll need to cancel, but a 500 for this API usually means a temporary problem fetching that last `continuationUrl` (the next 10 items). so the next time it tries the same continuationUrl.

I'm also going to test to see if the API is blacklisting IPs or if there is too much demand. It might not be a coincidence that your error happened around the same time I announced the release.
